File: language/java_study/javaLearn/src/main/java/com/iamhotcenplotz/chapter27/example/Example01_.java
package com.iamhotcenplotz.chapter27.example;

import java.util.regex.Matcher;
import java.util.regex.Pattern;

/**
 * @author: Der Hotcenplotz
 * @version： 1.0
 * @date: 2025-04-30 17:41
 * @description: 验证输入的是否全为汉字
 */


public class Example01_ {
    public static void main(String[] args) {
//        String content = "张三同学 Hello how are you 张三老师 find thank you 张三靓仔 and you i am find too";
        String content = "张三同学张三老师张三靓仔";
        String regStr = "^[\u0391-\uffe5]+$"; //汉字编码范围

        Pattern compile = Pattern.compile(regStr);
        Matcher matcher = compile.matcher(content);
        if(matcher.find()) {
            System.out.println("good");

        } else {
            System.out.println("bad");
        }
    }
}

File: language/scala_study/ScalaLearn/src/main/scala/com/hotcenplotz/chapter08/trait_/examples/Example01_.scala
package com.hotcenplotz.chapter08.trait_.examples

/**
 * @author: Der Hotcenplotz
 * @version： 1.0 
 * @date: 2025-05-13 06:43 
 * @description: trait 基本示例代码
 *
 * */
object Example01_ {
    
}
// trait
trait Trait01 {
    // 定义一个规范
    def getConnect()
    
}


// 六个案例类关系
class A {}
class B extends A {}
class C extends A with Trait01 {
    override def getConnect(): Unit = println("MySQL Connected")
}

class D {}
class E extends D {}
class F extends D with Trait01 {
    override def getConnect(): Unit = println("PostgreSQL Connected")
}